internal/security/secretstore/init.go

```diff
@@ -80,6 +83,17 @@ func (b *Bootstrap) BootstrapHandler(ctx context.Context, _ *sync.WaitGroup, _ s
 	vaultHost := fmt.Sprintf("%s:%v", configuration.SecretService.Server, configuration.SecretService.Port)
 	intervalDuration := time.Duration(b.vaultInterval) * time.Second
 	vc := secretstoreclient.NewSecretStoreClient(lc, req, vaultScheme, vaultHost)
+	pipedHexReader := pipedhexreader.NewPipedHexReader()
+	kdf := kdf.NewKdf(fileOpener, configuration.SecretService.TokenFolderPath, sha256.New)
+	vmkEncryption := NewVMKEncryption(fileOpener, pipedHexReader, lc, kdf)
+
+	if err := vmkEncryption.LoadIKM(os.Getenv("IKM_HOOK")); err != nil {
```

Comment:
mmm I think this code would be safer when written as

```suggestion
	err := vmkEncryption.LoadIKM(os.Getenv("IKM_HOOK"))
	defer vmkEncryption.WipeIKM()
	if err != nil {
```

this way even if we encounter a late error, if we had sensitive data in memory, we ensure it is wiped.

Reply:
rewritten as suggested.